config/boards/rockpro64.conf

```diff
@@ -10,29 +10,59 @@ BOOT_LOGO="desktop"
 BOOT_SCENARIO="blobless"
 BOOT_SUPPORT_SPI=yes
 
-# u-boot 2023.01 for rockpro64
-BOOTBRANCH_BOARD="tag:v2023.01"
-BOOTPATCHDIR="v2023.01"
+# u-boot v2024.04-rc4 for rockpro64
+BOOTBRANCH_BOARD="tag:v2024.04-rc4"
+BOOTPATCHDIR="v2024.04"
+
+# Include fw_setenv, configured to point to the correct spot on the SPI Flash
+PACKAGE_LIST_BOARD="libubootenv-tool" # libubootenv-tool provides fw_printenv and fw_setenv, for talking to U-Boot environment
 
 function add_host_dependencies__new_uboot_wants_python3_rockpro64() {
 	declare -g EXTRA_BUILD_DEPS="${EXTRA_BUILD_DEPS} python3-pyelftools" # @TODO: convert to array later
 }
 
-function post_family_config__hacks_for_new_uboot_rockpro64() {
-	display_alert "$BOARD" "using ATF (blobless) for 2023.01 u-boot" "info"
+function post_family_config__use_mainline_uboot_rockpro64() {
 	# Use latest lts 2.8 ATF
-	ATFBRANCH='tag:lts-v2.8.8'
+	ATFBRANCH='tag:lts-v2.8.16'
 	ATFPATCHDIR="atf-rockchip64" # patches for logging etc
-	# With recent-enough u-boot and ATF.
-	# bl31.elf is copied directly from ATF build dir to uboot dir anyway.
-	UBOOT_TARGET_MAP="BL31=bl31.elf;;idbloader.img u-boot.itb tpl/u-boot-tpl.bin spl/u-boot-spl.bin u-boot.itb rkspi_loader.img"
+	display_alert "$BOARD" "using ATF (blobless) ${ATFBRANCH} for ${BOOTBRANCH_BOARD} u-boot" "info"
+	# bl31.elf is copied directly from ATF build dir to uboot dir (by armbian u-boot build system)
+	UBOOT_TARGET_MAP="BL31=bl31.elf;;u-boot-rockchip.bin u-boot-rockchip-spi.bin"
+
+	# Ignore most of the rockchip64_common stuff, we're using binman here which does all the work already
+	unset uboot_custom_postprocess write_uboot_platform write_uboot_platform_mtd
+
+	function write_uboot_platform() {
+		dd "if=$1/u-boot-rockchip.bin" "of=$2" seek=64 conv=notrunc status=none
+	}
+
+	function write_uboot_platform_mtd() {
+		flashcp -v -p "$1/u-boot-rockchip-spi.bin" /dev/mtd0
+	}
 }
 
 function post_config_uboot_target__extra_configs_for_rockpro64() {
 	# Taken from https://gitlab.manjaro.org/manjaro-arm/packages/core/uboot-rockpro64/-/blob/master/PKGBUILD
-	display_alert "$BOARD" "u-boot configs for 2023.01 u-boot config" "info"
+	display_alert "$BOARD" "u-boot configs for ${BOOTBRANCH_BOARD} u-boot config" "info"
 	run_host_command_logged scripts/config --set-val CONFIG_OF_LIBFDT_OVERLAY "y"
 	run_host_command_logged scripts/config --set-val CONFIG_SPL_MMC_SDHCI_SDMA "n"
```

Comment:
This should not be needed with `v2023.07-rc3` or newer, workaround was added with https://github.com/u-boot/u-boot/commit/3b804b370dd8027a6d450c447247855a149ea2ec

Reply:
Thanks, removed!